framework/src/media/FileOutputDataSource.cpp

```diff
@@ -89,13 +95,13 @@ bool FileOutputDataSource::close()
 {
 	switch (getAudioType()) {
 		case AUDIO_TYPE_WAVE:
-			unsigned int fileSize;
+			long fileSize;
```

Comment:
uint16_t?

Reply:
At old line 94, expression `fileSize < 0` is always false.
We need long type variable to accept return value of `ftell`.

Reply:
How about this?
```c
long ret = ftell(mFp);
if (ret < 0) {
    meddbg(...);
    break;
}
unsigned int fileSize = (unsigned int)ret;
if (!utils::writeWavHeader(mFp, getChannels(), getSampleRate(), getPcmFormat(), fileSize)) {
    ...
}
```

Reply:
I think it is good idea.

Reply:
Fixed.